BrushUpCode/src/Coffee.java

```diff
@@ -0,0 +1,255 @@
+    private void buy(int dollar, int water, int milk, int coffeeBeans, int cups) {
+        Scanner enter = new Scanner(System.in);
+        System.out.println("What do you want to buy? 1) espresso 2) latte 3) cappuccino ? If mistaken you can go back by instruction  4) back");
+        int choice = enter.nextInt();
+        this.coffeeMachineState = State.MAKINGCOFFE;
+        switch (choice) {
+            case 1: {
+                System.out.println("Good Choice! ");
+                buyEspresso(dollar, water, milk, coffeeBeans, cups);
+                break;
+            }
+
+            case 2: {
+                System.out.println("Yummy Latte!");
+
+                buyLatte(dollar, water, milk, coffeeBeans, cups);
+                break;
+            }
+
+            case 3: {
+                System.out.println("Great choice!");
+                buyCappuccino(dollar, water, milk, coffeeBeans, cups);
+                break;
+            }
+            case 4: {
+                break;
+            }
+        }
+    }
+
+    //method for buying espresso(used in method buy)
+    private void buyEspresso(int dollar, int water, int milk, int coffeeBeans, int cups) {
+        int espressoWater = 250;
+        int espressoCoffee = 16;
+        int espressoMoney = 4;
+        if (water >= espressoWater && coffeeBeans >= espressoCoffee && cups >= 1) {
+            System.out.println("I have enough ingredients to make your coffee! \n Here is your coffee");
+            water -= espressoWater;
+            setWater(water);
+            coffeeBeans -= espressoCoffee;
+            setCoffeeBeans(coffeeBeans);
+            dollar += espressoMoney;
+            setDollar(dollar);
+            cups -= 1;
+            setCups(cups);
+
+
+        } else {
+            if (water < espressoWater)
+                System.out.println("not enough water");
+            else if (coffeeBeans < espressoCoffee)
+                System.out.println("not enough coffee");
+            else if (cups == 0)
+                System.out.println("not enough cups");
+        }
+
+    }
+
+    //method for buying Latte(used in method buy)
+    private void buyLatte(int dollar, int water, int milk, int coffeeBeans, int cups) {
+        int latteWater = 350;
+        int latteCoffee = 20;
+        int latteMilk = 75;
+        int latteMoney = 7;
+        if (water >= latteWater && coffeeBeans >= latteCoffee && cups >= 1 && milk >= latteMilk) {
+            System.out.println("I have enough ingredients to make your coffee! \n Here is your coffee");
+            water -= latteWater;
+            setWater(water);
+            milk -= latteMilk;
+            setMilk(milk);
+            coffeeBeans -= latteCoffee;
+            setCoffeeBeans(coffeeBeans);
+            dollar += latteMoney;
+            setDollar(dollar);
+            cups -= 1;
+            setCups(cups);
+        } else {
+            if (water < latteWater)
```

Comment:
this part of function in buyLatte, buyCappuccino, buyEspresso is same, can you fix it ?

Reply:
it's not the same, the variables are different. latteWater is not the same as espressoWater.  Please let me know what should I do to make it more efficient.

Reply:
All this fields is integer yes? can we make some function which is check this stuff and print "not enough water", "not enough coffee" ... etc

Reply:
I guess I fixed it